File: docs/code-examples/disconnected_space.rs
//! Disconnect two spaces.

use rerun::{DisconnectedSpace, Points3D, RecordingStreamBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) =
        RecordingStreamBuilder::new("rerun_example_disconnected_space").memory()?;

    // These two points can be projected into the same space..
    rec.log("world/room1/point", &Points3D::new([(0.0, 0.0, 0.0)]))?;
    rec.log("world/room2/point", &Points3D::new([(1.0, 1.0, 1.0)]))?;

    // ..but this one lives in a completely separate space!
    rec.log("world/wormhole", &DisconnectedSpace::new(true))?;
    rec.log("world/wormhole/point", &Points3D::new([(2.0, 2.0, 2.0)]))?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}

File: docs/snippets/all/disconnected_space.py
"""Disconnect two spaces."""
import rerun as rr

rr.init("rerun_example_disconnected_space", spawn=True)

# These two points can be projected into the same space..
rr.log("world/room1/point", rr.Points3D([[0, 0, 0]]))
rr.log("world/room2/point", rr.Points3D([[1, 1, 1]]))

# ..but this one lives in a completely separate space!
rr.log("world/wormhole", rr.DisconnectedSpace())
rr.log("world/wormhole/point", rr.Points3D([[2, 2, 2]]))
